### PR TITLE
Dubey | 000 | Refactor to work for multiple variants

### PR DESCRIPTION
**Pull request by mddubey**
Changes for HIU to work with multiple variants

### CODE REVIEW

File: src/main/java/in/org/projecteka/hiu/valuseset/ValueSetController.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+import reactor.core.publisher.Mono;
+
+@RestController
+@AllArgsConstructor
+public class ValueSetController {
+    private ValueSetResource valueSetResource;
+
+    @GetMapping(value = "ValueSet", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (angshu) [inline src/main/java/in/org/projecteka/hiu/valuseset/ValueSetController.java:14]: you probably have to configure/map the HIU UI api call as well

File: src/main/java/in/org/projecteka/hiu/valuseset/ValueSetResource.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import org.apache.commons.io.IOUtils;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Component;
+
+import java.nio.charset.Charset;
+
+@Component
+public class ValueSetResource implements InitializingBean {
```
Comment (angshu) [inline src/main/java/in/org/projecteka/hiu/valuseset/ValueSetResource.java:12]: The ConceptValidator class already does the loading. Maybe we can refactor the ConceptValidator to use this class instead?

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ hiu:
   name: ${HIU_NAME}
   dataPushUrl: ${DATA_PUSH_URL}
   consentNotificationUrl: ${CONSENT_NOTIFICATION_URL}
-  valueSets: classpath:/static/ValueSet/valuesets.json
+  valueSets: classpath:/static/ValueSet/ncg-valuesets.json
```
Comment (angshu) [inline src/main/resources/application.yml]: should this not be driven by configuration? we have the a default value (am ok with making ref to ndhm defined hiTypes and purposes).

File: src/main/java/in/org/projecteka/hiu/valuseset/ValueSetResource.java (hunk 2 above)
Comment (mddubey) [reply src/main/java/in/org/projecteka/hiu/valuseset/ValueSetResource.java:12]: Sure!

File: src/main/java/in/org/projecteka/hiu/valuseset/ValueSetController.java (hunk 1 above)
Comment (mddubey) [reply src/main/java/in/org/projecteka/hiu/valuseset/ValueSetController.java:14]: On that.

**Review by angshu — APPROVED** [review]

**Pull request merged**